processor/batchprocessor/config.go

```diff
@@ -1,68 +1,69 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
 
-package batchprocessor // import "go.opentelemetry.io/collector/processor/batchprocessor"
+package batchprocessor
 
-import (
-	"errors"
-	"fmt"
-	"strings"
-	"time"
+import "encoding/json"
+import "fmt"
+import "time"
 
-	"go.opentelemetry.io/collector/component"
-)
-
-// Config defines configuration for batch processor.
+// Configuration parameters for the batch processor.
 type Config struct {
-	// Timeout sets the time after which a batch will be sent regardless of size.
-	// When this is set to zero, batched data will be sent immediately.
-	Timeout time.Duration `mapstructure:"timeout"`
+	// MetadataCardinalityLimit corresponds to the JSON schema field
+	// "metadata_cardinality_limit".
+	MetadataCardinalityLimit int `mapstructure:"metadata_cardinality_limit,omitempty"`
```

Comment:
I need to check whether having the `omitempty` is a problem. OTel doesn't use it much right now, but it looks like go-jsonschema always puts it in.

Reply:
I don't think omitempty is used by mapstructure at all.